### PR TITLE
feat:delete  redundant configurations

### PR DESCRIPTION
**Pull request by constwz**
Simplify block syncer and metadata configuration
Delete the Backup DB

### CODE REVIEW

**Review by BarryTong65 — APPROVED** [review]

File: base/gfspapp/app_options.go [hunk 1]
```diff
@@ -305,42 +305,42 @@ func DefaultGfBsDBOption(app *GfSpBaseApp, cfg *gfspconfig.GfSpConfig) error {
...
+			//}
+			//if val, ok := os.LookupEnv(bsdb.BsDBSwitchedPasswd); ok {
+			//	cfg.BsDBBackup.Passwd = val
+			//}
+			//if val, ok := os.LookupEnv(bsdb.BsDBSwitchedAddress); ok {
+			//	cfg.BsDBBackup.Address = val
+			//}
+			//if val, ok := os.LookupEnv(bsdb.BsDBSwitchedDatabase); ok {
+			//	cfg.BsDBBackup.Database = val
+			//}
```
Comment (mliu233) [inline base/gfspapp/app_options.go:325]: If we are sure these are not needed, we shall remove the code other than comment them out.

File: base/gfspapp/app_options.go [hunk 2]
```diff
@@ -305,42 +305,42 @@ func DefaultGfBsDBOption(app *GfSpBaseApp, cfg *gfspconfig.GfSpConfig) error {
...
-			}
+			//bsDBBlockSyncerBackUp, err := bsdb.NewBsDB(cfg, true)
+			//if err != nil {
+			//	log.Panicw("failed to new backup bsdb", "error", err)
+			//	return
+			//}
 
 			app.gfBsDBMaster = bsDBBlockSyncerMaster
-			app.gfBsDBBackup = bsDBBlockSyncerBackUp
+			//app.gfBsDBBackup = bsDBBlockSyncerBackUp
```
Comment (mliu233) [inline base/gfspapp/app_options.go:343]: ditto.

File: base/gfspapp/app_test.go [hunk 3]
```diff
@@ -83,11 +83,11 @@ func TestGfSpBaseApp_GfBsDBMaster(t *testing.T) {
...
-func TestGfSpBaseApp_GfBsDBBackup(t *testing.T) {
-	g := setup(t)
-	result := g.GfBsDBBackup()
-	assert.Nil(t, result)
-}
+//func TestGfSpBaseApp_GfBsDBBackup(t *testing.T) {
+//	g := setup(t)
+//	result := g.GfBsDBBackup()
+//	assert.Nil(t, result)
+//}
```
Comment (mliu233) [inline base/gfspapp/app_test.go:91]: ditto

File: modular/metadata/metadata_options.go [hunk 4]
```diff
@@ -82,42 +72,41 @@ func DefaultMetadataOptions(metadata *MetadataModular, cfg *gfspconfig.GfSpConfi
...
-}
+//func switchDB(flag bool, cfg *gfspconfig.GfSpConfig, metadata *MetadataModular) {
+//	if flag {
+//		metadata.baseApp.SetGfBsDB(metadata.baseApp.GfBsDBMaster())
+//	} else {
+//		metadata.baseApp.SetGfBsDB(metadata.baseApp.GfBsDBBackup())
+//	}
+//	cfg.Metadata.IsMasterDB = flag
+//	log.Info("db switched successfully")
+//}
```
Comment (mliu233) [inline modular/metadata/metadata_options.go:112]: ditto.

**Review by mliu233 — COMMENTED** [review]
Please also update your pr description so reviewers & future code readers know what is going on with this pr.

File: base/gfspapp/app_options.go (hunk 1 above)
Comment (constwz) [reply base/gfspapp/app_options.go:325]: Delete the configuration first, leave the code, and add it back later depending on the situation

Comment (ruojunm) [reply base/gfspapp/app_options.go:325]: +1
Const, Can you just remove them?
If needed, you can find them via git histories with searching commit messages.

File: base/gfspapp/app_options.go (hunk 2 above)
Comment (ruojunm) [reply base/gfspapp/app_options.go:343]: +1

File: modular/metadata/metadata_options.go [hunk 5]
```diff
@@ -59,21 +50,20 @@ func DefaultMetadataOptions(metadata *MetadataModular, cfg *gfspconfig.GfSpConfi
...
-	if cfg.Metadata.IsMasterDB {
-		metadata.baseApp.SetGfBsDB(metadata.baseApp.GfBsDBMaster())
-	} else {
-		metadata.baseApp.SetGfBsDB(metadata.baseApp.GfBsDBBackup())
-	}
+	//	if cfg.Metadata.IsMasterDB {
+	metadata.baseApp.SetGfBsDB(metadata.baseApp.GfBsDBMaster())
+	//} else {
+	//	metadata.baseApp.SetGfBsDB(metadata.baseApp.GfBsDBBackup())
+	//}
```
Comment (ruojunm) [inline modular/metadata/metadata_options.go:57]: please remove useless code

File: modular/metadata/metadata_options.go [hunk 6]
```diff
@@ -59,21 +50,20 @@ func DefaultMetadataOptions(metadata *MetadataModular, cfg *gfspconfig.GfSpConfi
...
 	BsModules = cfg.BlockSyncer.Modules
-	BsEnableDualDB = cfg.BlockSyncer.EnableDualDB
 	BsWorkers = cfg.BlockSyncer.Workers
 	ChainID = cfg.Chain.ChainID
 	ChainAddress = cfg.Chain.ChainAddress
 	SpOperatorAddress = cfg.SpAccount.SpOperatorAddress
 	GatewayDomainName = cfg.Gateway.DomainName
 
-	startDBSwitchListener(time.Second*time.Duration(cfg.Metadata.BsDBSwitchCheckIntervalSec), cfg, metadata)
+	//startDBSwitchListener(time.Second*time.Duration(cfg.Metadata.BsDBSwitchCheckIntervalSec), cfg, metadata)
```
Comment (ruojunm) [inline modular/metadata/metadata_options.go:66]: ditto

File: modular/metadata/metadata_options.go [hunk 7]
```diff
@@ -82,42 +72,41 @@ func DefaultMetadataOptions(metadata *MetadataModular, cfg *gfspconfig.GfSpConfi
...
+//		for range dbSwitchTicker.C {
+//			checkSignal(cfg, metadata)
+//		}
+//	}()
+//}
+
+//func checkSignal(cfg *gfspconfig.GfSpConfig, metadata *MetadataModular) {
+//	// check if there is a signal from block syncer database to switch the database
+//	signal, err := metadata.baseApp.GfBsDBMaster().GetSwitchDBSignal()
+//	if err != nil || signal == nil {
```
Comment (ruojunm) [inline modular/metadata/metadata_options.go:91]: ditto

File: modular/metadata/metadata_sp_service.go [hunk 8]
```diff
@@ -127,8 +127,8 @@ func (r *MetadataModular) GfSpGetStatus(ctx context.Context, req *types.GfSpGetS
 		BsDefaultCharacterSetName: defaultCharacterSetName,
 		BsDefaultCollationName:    defaultCollationName,
 		BsModules:                 BsModules,
-		BsEnableDualDb:            BsEnableDualDB,
-		BsWorkers:                 uint32(BsWorkers),
+		//BsEnableDualDb:            BsEnableDualDB,
```
Comment (ruojunm) [inline modular/metadata/metadata_sp_service.go:130]: ditto

**Review by ruojunm — CHANGES_REQUESTED** [review]

**Review by ruojunm — APPROVED** [review]

**Pull request merged**